modin/core/dataframe/pandas/dataframe/utils.py

```diff
@@ -0,0 +1,266 @@
+    t = num_partitions, and n = total_length.
+
+    Parameters
+    ----------
+    df : pandas.Dataframe
+        The masked dataframe to pick samples from.
+    column : str
+        The columns to pick quantiles from. Only the first column in the list will be used.
+    num_partitions : int
+        The number of partitions.
+    length : int
+        The total length.
+
+    Returns
+    -------
+    np.ndarray:
+        The samples for the partition.
+
+    Notes
+    -----
+    This sampling algorithm is inspired by TeraSort. You can find more information about TeraSort
+    and the sampling algorithm at https://www.cse.cuhk.edu.hk/~taoyf/paper/sigmod13-mr.pdf.
+    """
+    m = length / num_partitions
+    probability = (1 / m) * np.log(num_partitions * length)
+    return df[column].sample(frac=probability).to_numpy()
+
+
+def pick_pivots_from_samples_for_sort(
+    df: PandasDataframe,
+    samples: np.ndarray,
+    method: str = "linear",
+    key: Optional[Callable] = None,
+) -> np.ndarray:
+    """
+    Determine quantiles from the given samples.
+
+    This function takes as input the quantiles calculated over all partitions from
+    `sample_func` defined above, and determines a final NPartitions.get() quantiles
+    to use to roughly sort the entire dataframe. It does so by collating all the samples
+    and computing NPartitions.get() quantiles for the overall set.
+
+    Parameters
+    ----------
+    df : PandasDataframe
+        The Modin Dataframe calling this function.
+    samples : np.ndarray
+        The samples computed by ``get_partition_quantiles_for_sort``.
+    method : str, default: linear
+        The method to use when picking quantiles.
+    key : Callable, default: None
+        The key to use on the samples when picking pivots.
+
+    Returns
+    -------
+    np.ndarray
+        A list of overall quantiles.
+    """
+    # We don't call `np.unique` on the samples, since if a quantile shows up in multiple
+    # partition's samples, this is probably an indicator of skew in the dataset, and we
+    # want our final partitions to take this into account.
+    # We need to use numpy to concatenate the samples since the sample from each partition is
+    # a NumPy array, and we want one flattened array of samples.
+    all_pivots = np.concatenate(samples).flatten()
+    if key is not None:
+        all_pivots = key(all_pivots)
+    # We don't want to pick very many quantiles if we have a very small dataframe.
+    num_quantiles = len(df._partitions)
+    quantiles = [i / num_quantiles for i in range(1, num_quantiles)]
+    overall_quantiles = _find_quantiles(all_pivots, quantiles, method)
+    return overall_quantiles
+
+
+def split_partitions_using_pivots_for_sort(
+    modin_frame: PandasDataframe,
+    df: pandas.DataFrame,
+    column: str,
+    pivots: np.ndarray,
+    ascending: bool,
+    **kwargs: dict,
+) -> "tuple[pandas.DataFrame]":
```

Comment:
```suggestion
) -> "tuple[pandas.DataFrame, ...]":
```
otherwise you type-hint that you return a 1-element tuple (yeah, [hinting a tuple is different from hinting a list](https://stackoverflow.com/a/54747327)... I mean, "C" in "type hint" stands for "Consistency" 🤪 )